numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,92 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape, step=None):
+    """
+    Create rolling window views of the 2D array with the given shape.
+
+    For some cases, there may be more efficient approaches
+
+    Parameters
+    ----------
+    x : ndarray
+        Array to create rolling window views.
+    shape : sequence of int
+        The shape of the sliding window.
+    step: sequence of int, optional
+        The sliding step of view window for each dimension of input. Defaults to 1 on all dimensions.
```

Comment:
Reiterating my comment earlier - I don't think the `step` argument belongs here: https://github.com/numpy/numpy/issues/7753#issuecomment-374836132

Reply:
Personally, I think sliding_window(x, shape, step) is more efficient than sliding_window(x, shape)[::step], since the latter one requires creating view first, then take steps. This could be problematic in large input array. Also, the 'step' parameter (also mentioned as 'stepsize', 'stride') is suggested by the first comment([Here](https://github.com/numpy/numpy/issues/7753#issue-160868188). I think it would be handy to keep this optional parameter.

Reply:
> which could be problematic in large input array

The cost of a view is constant, and does not depend on the `.shape` of the array - so "large" is irrelevant.

Reply:
> Furthermore, the step argument is ambiguous - does it specify the step within a window, or the step between windows?

In deep learning world(where I came from),  step within a window is usually called 'dilation', step between windows is usually called 'stride'. However, in numpy 'strides' usually means step **by bytes** in each dimension. I agree that the naming causes confusion here, but I think 'step' (or whatever it called) is used quite frequently. Maybe we can come up with a better name?

Reply:
> Let's discuss the step argument in the issue, not in the implementation.

@eric-wieser I clarify the document by adding ' window shifts' to this parameter description. I think this could avoid ambiguity and this parameter is quite useful. Do you still think we should remove this parameter?